### PR TITLE
Add error page

### LINKED ISSUE #148: Describe errors and exceptions
In the usage/client section, from a user perspective, it would be useful to provide a description (and tips) about errors and exceptions raised while manipulating assets.

See: https://github.com/SubstraFoundation/substra-backend/blob/master/backend/substrapp/tasks/exceptions.json

@camillemarini Would you like to add any comment/example on this issue? I will open a draft PR as soon as possible on this!

### PR DESCRIPTION
**Pull request by natct10**
Fix #148, #152 & #154

Thx @celinejacques for the contribution!

### CODE REVIEW

**Review by jmorel — APPROVED** [review]

File: src/errors.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+## Command Line Interface
+
+[CLI documentation](https://github.com/SubstraFoundation/substra/blob/master/references/cli.md#summary)
+
+> Remember: you can use `substra --help` and `substra <command> --help` anytime!
+
+### substra config
+
+> `substra config --profile <profile> <url>`
+
```
Comment (RomainGoussault) [inline src/errors.md:12]: We may remove these lines since we don't provide any errors for this command?

File: src/errors.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+### Generate data sample
+
+Check the `generate_data_samples.py` script and the path to your dataset, the values of `N_TRAIN_DATA_SAMPLES` and `N_TEST_DATA_SAMPLES`.
+
+<details>
+<summary><b>FileExistsError</b></summary>
+This indicates that the `train_data_samples` & `test_data_samples` have already been generated in the `assets` folder. This means that you will need to remove it before re-generating data samples.
+</details>
+
```
Comment (RomainGoussault) [inline src/errors.md:104]: I believe this is specific to the titanic example? If so I would remove it. Or maybe put it in the readme of the example. What do you think @natct10?

File: src/errors.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+### Register dataset and objective
+
+Check the `add_dataset_objective.py` script and your assets (`DATASET`, `TEST_DATA_SAMPLES_PATHS`, `TRAIN_DATA_SAMPLES_PATHS`, `OBJECTIVE`, `METRICS_DOCKERFILE_FILES`).
+
+### Add an algorithm
+
+Check the `add_algo.py` script and your assets (`ALGO`, `ALGO_DOCKERFILE_FILES`).
+
```
Comment (RomainGoussault) [inline src/errors.md:112]: Same remark here.

File: src/errors.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+If after adding an algorithm it directly gets a `failed` status, it might be related to the `permissions`: please be sure to allow, in the dataset from the data provider org, the algo provider org to access the related dataset. You will also have to edit the algo permissions and to make it available for the data provider org.
+
+> Note: this is not required in debug mode
+
+</details>
+
+### Add a train tuple
+
+In debug mode, when you run the `add_train_tuple` method with large data, a Docker container is launched and then exit showing this last log `substratools.algo - launching training task` and no further stacktrace. This might be related to a lack of memory (use `docker stats`) and you tweak this limit by using a higher `shm_size` in the `spawner.py` file.
```
Comment (RomainGoussault) [inline src/errors.md:124]: typo: `add_traintuple`

File: src/errors.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+## Exhaustive list of exceptions
+
+If you want to go further, you can have a look to the [full list of exceptions](https://github.com/SubstraFoundation/substra-backend/blob/master/backend/substrapp/tasks/exceptions.json) handled by the backend. This can provide you with some insight about the issue you are facing.
```
Comment (RomainGoussault) [inline src/errors.md:128]: I would remove this part since the user does not have access to these exceptions or these error codes.

**Review by RomainGoussault — COMMENTED** [review]
Small comments, this error page will help a lot @natct10 !

File: src/errors.md (hunk 1 above)
Comment (natct10) [reply src/errors.md:12]: In fact, you get the `HTTPConnectionPool` error if you don't use the `--profile` param, that's why I plced this reminder. what do you think about it?

File: src/errors.md (hunk 2 above)
Comment (natct10) [reply src/errors.md:104]: It is indeed based on Titanic example but I though it was a good illustration of the `FileExistsError` error. Are you sure to want to remove it?

File: src/errors.md (hunk 3 above)
Comment (natct10) [reply src/errors.md:112]: Same

File: src/errors.md (hunk 4 above)
Comment (natct10) [reply src/errors.md:124]: oh thx!

File: src/errors.md (hunk 1 above)
Comment (RomainGoussault) [reply src/errors.md:12]: I thought the 503 was for the substra login only. (maybe the rendering will be different on the doc website)

Comment (RomainGoussault) [reply src/errors.md:12]: Actually both the 503 error and the HTTPConnectionPool error can be encountered with any CLI command?

File: src/errors.md (hunk 2 above)
Comment (RomainGoussault) [reply src/errors.md:104]: `FileExistsError` is a python error but unrelated to Substra?
It appends there --> https://github.com/SubstraFoundation/substra/blob/master/examples/titanic/scripts/generate_data_samples.py#L66 ?

File: src/errors.md (hunk 5 above)
Comment (RomainGoussault) [reply src/errors.md:128]: It will not be fixed as we are thinking about a new way to display errors and logs.

Comment (natct10) [reply src/errors.md:128]: Ok noted

File: src/errors.md (hunk 2 above)
Comment (natct10) [reply src/errors.md:104]: Yep this is indeed a python error you can face when you have already generated samples. I can remove this if you prefer

File: src/errors.md (hunk 1 above)
Comment (natct10) [reply src/errors.md:12]: Yes, but I saw more frequently on the first steps.

**Review by RomainGoussault — APPROVED** [review]

**Pull request merged**